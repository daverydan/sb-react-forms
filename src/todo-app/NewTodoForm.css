.NewTodoForm label {
  display: block;
  margin-bottom: 10px;
}

.NewTodoForm input {
  margin-bottom: 10px;
}
