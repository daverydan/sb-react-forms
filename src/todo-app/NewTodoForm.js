import React, { useState } from "react";
import "./NewTodoForm.css";

const NewTodoForm = ({ addTodo }) => {
  const INITIAL_STATE = { todo: "" };
  const [form, setForm] = useState(INITIAL_STATE);
  const handleChange = (e) => {
    const { name, value } = e.target;
    setForm({ [name]: value });
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo(form.todo);
    setForm(INITIAL_STATE);
  };
  return (
    <form className="NewTodoForm" onSubmit={handleSubmit}>
      <label htmlFor="todo">Add Task</label>
      <input
        type="text"
        name="todo"
        id="todo"
        value={form.todo}
        onChange={handleChange}
        autoFocus
      />
      <button>Add</button>
    </form>
  );
};

export default NewTodoForm;
