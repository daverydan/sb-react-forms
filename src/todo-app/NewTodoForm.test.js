import React from "react";
import { render } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("renders w/out crashing", () => {
  render(<NewTodoForm />);
});

test("it matches snapshot", () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});
