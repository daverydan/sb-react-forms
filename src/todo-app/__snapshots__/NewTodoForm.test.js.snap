// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <label
      for="todo"
    >
      Add Task
    </label>
    <input
      id="todo"
      name="todo"
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
